File: javasample/src/main/java/cleanTests/Executor.java
package cleanTests;

import java.util.Optional;

public class Executor {

    private final String id;
    private String group;

    public Executor(String id) {
        this.id = id;
    }

    public Executor(String id, String group) {
        this.id = id;
        this.group = group;
    }

    public String getID() {
        return this.id;
    }

    public Optional<String> getGroup() {
        return Optional.ofNullable(this.group);
    }

    public Executor setGroup(String group) {
        this.group = group;
        return this;
    }

    public boolean isInGroup(String group) {
        return group.equalsIgnoreCase(this.group);
    }

}

File: cppsample/src/cleantests/cpp/executor.cpp
#include "executor.h"

Executor::Executor(std::string id) {
    id_ = id;
}

Executor::Executor(std::string id, std::string group) {
    id_ = id;
    group_ = group;
}

std::string Executor::getGroup() {
    return group_;
}

std::string Executor::getID() {
    return id_;
}

bool Executor::isInGroup(const std::string group) {
    return group_ == group;
}